mne_icalabel/features/topomap.py

```diff
@@ -0,0 +1,141 @@
+        picks = _picks_to_idx(ica.n_components_, picks)
+    data = np.dot(
+        ica.mixing_matrix_[:, : ica.n_components_].T,
+        ica.pca_components_[: ica.n_components_],
+    )
+    # Create an empty array of size (len(picks), 64, 64) for the topomap
+    topomaps = np.zeros((len(picks), res, res))
+    for i, component in enumerate(picks):
+        topo = np.flipud(
+            get_topomap(
+                data[component, :], ica.info, res, outlines, image_interp, border, extrapolate
+            )
+        )
+        # Set NaN values to 0
+        np.nan_to_num(topo, nan=0.0, copy=False)
+        # Standardize the values
+        topomaps[i, :, :] = topo / np.max(np.abs(topo))
+    return topomaps  # topographic map array for all the picked components (len(picks), 64, 64)
+
+
+@fill_doc
+def get_topomap(
+    data: NDArray[float],
+    pos: Info,
+    res: int = 64,
+    outlines: Optional[str] = "head",
+    image_interp: str = _INTERPOLATION_DEFAULT,  # 'cubic'
+    border: Union[float, str] = _BORDER_DEFAULT,  # 'mean'
+    extrapolate: str = _EXTRAPOLATE_DEFAULT,  # 'head'
+):
+    """Generate a scalp topographic map (n_pixels, n_pixels).
+
+    Parameters
+    ----------
+    data : array of shape (n_channels,)
+        The data points used to generate the topographic map.
+    pos : Info
+        Instance of `mne.Info` with the montage associated with the (n_channels,) points.
+    %(res_topomap)s
+    %(outlines_topomap)s
+    %(image_interp_topomap)s
+    %(border_topomap)s
+    %(extrapolate_topomap)s
+
+    Returns
+    -------
+    topomap : array of shape (n_pixels, n_pixels)
+        Topographic map array.
+    """
+    _check_mne_version()
+
+    picks = _pick_data_channels(pos, exclude=())  # pick only data channels
+    pos = pick_info(pos, picks)
+    ch_type = _get_channel_types(pos, unique=True)
+
+    if len(ch_type) > 1:
+        raise ValueError("Multiple channel types in Info structure.")
+    elif len(pos["chs"]) != data.shape[0]:
+        raise ValueError(
+            "The number of channels in the Info object and in the data array do not match."
+        )
+    else:
+        ch_type = ch_type.pop()
+
+    picks = list(range(data.shape[0]))
+    sphere = np.array([0.0, 0.0, 0.0, 0.095])
+
+    # inferring (x, y) coordinates form mne.Info instance
+    pos = _find_topomap_coords(pos, picks=picks, sphere=sphere)
+    extrapolate = _check_extrapolate(extrapolate, ch_type)
+
+    # interpolation, valid only for MNE ≥ 1.1
+    outlines = _make_head_outlines(sphere, pos, outlines, (0.0, 0.0))
+    extent, Xi, Yi, interp = _setup_interp(pos, res, image_interp, extrapolate, outlines, border)
+    interp.set_values(data)
+    return interp.set_locations(Xi, Yi)()  # Zi, topomap of shape (n_pixels, n_pixels)
+
+
+def _check_mne_version():
+    """Check that MNE version is above 1.1."""
+    if not check_version("mne", "1.1"):
+        raise RuntimeError("Topographic feature is only available for MNE ≥ 1.1")
```

Comment:
Should this be in utils, and maybe add an argument to make it generalize for any version?

Reply:
Usually, I would say yes, but this code should be removed once we bump the MNE requirement from 1.0 to 1.1. 
I think keeping this 'temporary' code in the same file makes this future step easier, even if in this case I don't foresee this req. bump to happen anytime soon.

I don't think we need a general case for this function, we won't use it anyway and we should support all the versions of MNE included in the requirements.

By the way, this topographic feature requires MNE 1.1 because of recent changes and bug fixes in the interpolation. c.f. https://github.com/mne-tools/mne-python/pull/10579 and https://github.com/mne-tools/mne-python/pull/10617

Reply:
Okay fair. I'm in favor of just automatically going to only support v1.1+ for mne-icalabel v0.3 since we're in super-fast dev mode.

Maybe add a `#TODO` comment then to remove this when v1.1 is released?

Reply:
That works for me!